### PR TITLE
Add 24-hour time format support for weather

### PR DESCRIPTION
**Pull request by tarikcoskun**
Added `hour-format` option to the configuration so that the forecast could be in 12-hour or 24-hour format, with 12-hour being the default.

image

### CODE REVIEW

File: internal/widget/weather.go [hunk 1]
```diff
@@ -14,17 +14,26 @@ type Weather struct {
...
 	widget.withTitle("Weather").withCacheOnTheHour()
-	widget.TimeLabels = timeLabels
+
+	if widget.HourFormat == "" || widget.HourFormat == "12h" {
+		widget.TimeLabels = timeLabels12h
+	} else if widget.HourFormat == "24h" {
+		widget.TimeLabels = timeLabels24h
+	} else if widget.Units != "12h" && widget.Units != "24h" {
+		return fmt.Errorf("invalid hour format '%s' for weather, must be either 12h or 24h", widget.HourFormat)
+	}
```
Comment (Horib) [inline internal/widget/weather.go:36]: Why not use an if-else statement to simplify the code?

```
if widget.HourFormat == "24h" {
    widget.TimeLabels = timeLabels24h
} else {
    widget.TimeLabels = timeLabels12h
}
```

An other suggestion could be 

```
if widget.HourFormat == "12h" {
    widget.TimeLabels = timeLabels12h
}

if widget.HourFormat == "24h" {
    widget.TimeLabels = timeLabels24h
}

if widget.HourFormat != "12h" && widget.HourFormat != "24h" {
    return fmt.Errorf("invalid hour format '%s' for weather, must be either 12h or 24h", widget.HourFormat)
}
```

Comment (tarikcoskun) [reply internal/widget/weather.go:36]: Certainly, that would work. I wanted to stay consistent with the owner's style with the error when the value is invalid. I just noticed though, the mistake on the 34th line checking for `widget.Units`. I will wait for the owner's comments for further changes.

**Comment by svilenmarkov** [comment]
Hey, thanks for contributing!

Don't worry about the units check, I've sorted it out.

File: internal/widget/weather.go [hunk 2]
```diff
@@ -14,17 +14,26 @@ type Weather struct {
...
+	HourFormat   string          `yaml:"hour-format"`
 	Units        string          `yaml:"units"`
 	Place        *feed.PlaceJson `yaml:"-"`
 	Weather      *feed.Weather   `yaml:"-"`
 	TimeLabels   [12]string      `yaml:"-"`
 }
 
-var timeLabels = [12]string{"2am", "4am", "6am", "8am", "10am", "12pm", "2pm", "4pm", "6pm", "8pm", "10pm", "12am"}
+var timeLabels12h = [12]string{"2am", "4am", "6am", "8am", "10am", "12pm", "2pm", "4pm", "6pm", "8pm", "10pm", "12am"}
+var timeLabels24h = [12]string{"02:00", "04:00", "06:00", "08:00", "10:00", "12:00", "14:00", "16:00", "18:00", "20:00", "22:00", "24:00"}
```
Comment (p5) [inline internal/widget/weather.go:25]: Should this have been `00:00` rather than `24:00`?

**Pull request merged**